Replicator/c4Socket.cc

```diff
@@ -37,6 +37,14 @@ using namespace fleece;
 using namespace litecore;
 using namespace litecore::net;
 
+#define catchAndClose() \
+    catch (const std::exception &x) { \
+        closeWithException(x); \
+    } catch (...) { \
```

Comment:
We have no non-`std::exception` exceptions, so this isn't needed.

Reply:
Interestingly some Windows errors manifest as exceptions that are not based on std::exception where on other platforms they would be signals.  This is the only way to catch such errors.  Whether or not that is useful, is another story.